tests/integration/test_poly_def.py

```diff
@@ -0,0 +1,101 @@
+from collections.abc import Callable
+
+from guppylang.decorator import guppy
+from guppylang.module import GuppyModule
+from guppylang.prelude.builtins import array
+
+
+def test_id(validate):
+    module = GuppyModule("test")
+    T = guppy.type_var("T", module=module)
```

Comment:
I'd prefer to keep it a `guppy` method because most users will work with implicit modules. Explicit modules are only really used in tests, so imo it's fine to be more verbose there